wagtail/wagtailcore/models.py

```diff
@@ -1084,6 +1084,7 @@ def as_page_object(self):
 
         # Override the possibly-outdated tree parameter fields from this revision object
         # with up-to-date values
+        obj.pk = self.page.pk
```

Comment:
This is required as we hit Djangos unsaved instance check (outside of modelcluster as that bypasses it). Might be worth investigating...
https://docs.djangoproject.com/en/dev/releases/1.8/#assigning-unsaved-objects-to-relations-raises-an-error